src/ORM/Table.php

```diff
@@ -2531,9 +2531,11 @@ public function __get($property)
         $association = $this->_associations->get($property);
         if (!$association) {
             throw new RuntimeException(sprintf(
-                'Table "%s" is not associated with "%s"',
-                get_class($this),
-                $property
+                'Undefined property `%s`. ' .
+                'You have not defined an association between `%s` and `%s` tables.',
```

Comment:
this is not clear now 

I think this should tell to user that which is base table  name and which target table

Reply:
How about now?

Reply:
I think the information is provided by the exception itself:

$this->Articles->Tags
You have not set the association between Articles and Tags.

Working with a team new to cakephp, I think the people confused at this point are always going to be confused until they spend a little more time with cakephp. Associations in general are confusing!  Most of the emails I received early on was about queries not working because I forgot to set the right binding key on associations/etc and the SQL error makes no sense to them.

Reply:
@markstory or sync with `getAssociation()`
https://github.com/cakephp/cakephp/blob/32d9d38cca72e1795b940a91e1154c8253dc7e56/src/ORM/Table.php#L1007

Reply:
So the two error messages need to be the same? Could that message be updated separately?

Reply:
So the two error messages need to be the same? Could that message be updated separately?

Reply:
So the two error messages need to be the same? Could that message be updated separately?

Reply:
>So the two error messages need to be the same?

no,  just a suggestion 
now is good :+1: